### ISSUE TITLE
Make OPT_BINARY_PROTOCOL configurable

### ISSUE THREAD

**Issue by danieledangeli** [php-cache/issues#106]
Hello guys,
we're using your library for the cache pool.
We have exprienced some problems with the OPT_BINARY_PROTOCOL and sadly realized that you set this option under the hood. 

Would be really useful if this option could be configurable (like if I want disable it by passing an option to the  __construct method).

 It does make sense?

Followoing the piece of code I'm talking about

```php
**
 * @author Aaron Scherer <aequasi@gmail.com>
 * @author Tobias Nyholm <tobias.nyholm@gmail.com>
 */
class MemcachedCachePool extends AbstractCachePool implements HierarchicalPoolInterface
{
    use HierarchicalCachePoolTrait;
    use TagSupportWithArray;

    /**
     * @type \Memcached
     */
    protected $cache;

    /**
     * @param \Memcached $cache
     */
    public function __construct(\Memcached $cache)
    {
        $this->cache = $cache;
        $this->cache->setOption(\Memcached::OPT_BINARY_PROTOCOL, true);
    }

...
}
```

**Comment by Nyholm** [comment]
Hey. 

Im not sure at the moment why we set this option. Of course it is better for performance but I believe it also solved some issues. Anyhow, since I do not have a clear reson, I would be happy if you send a PR allowing people to configure this behavior.

**Comment by danieledangeli** [comment]
Yes of course, if it make sense I can do it. I had to rewrite the MemcacheCachePool because this option was causing some problems for us

**Comment by Nyholm** [comment]
Tell me more. What issues did you have?

**Comment by prisis** [comment]
@danieledangeli any news?

**Comment by prisis** [comment]
No feedback was provided so im closing this.

**Comment by rgershkovich** [comment]
One of the issues – mcrouter does not support binary memcached protocol:
https://github.com/facebook/mcrouter/issues/6

Although a clear limitation on mcrouter side, makes it impossible to use together with Memcached php-cache.

With OPT_BINARY_PROTOCOL set to false all works correctly.

**Comment by cryptiklemur** [comment]
Thanks for reporting this. [cache/memcached-adapter 3.0.0 accepts an array of Memcached options](https://github.com/php-cache/memcached-adapter/blob/1eb05db3b17b52e26a9fe1d80cc5048a61e8a8a1/MemcachedCachePool.php#L31-L43) and applies them over the default OPT_BINARY_PROTOCOL => true. [The tests](https://github.com/php-cache/memcached-adapter/blob/1eb05db3b17b52e26a9fe1d80cc5048a61e8a8a1/Tests/MemcachedCachePoolTest.php#L24-L53) cover multiple options and explicitly disabling the binary protocol. I'm closing this as resolved. If you run into it again on the current release, please reopen the issue with a reproducer.
